src/com/hms_networks/americas/sc/taginfo/TagInfoManager.java

```diff
@@ -90,11 +89,19 @@ public static synchronized void refreshTagList() throws IOException {
        * Add received byte to array (if not new line,
        * carriage return or end of stream)
        */
-      if (currentByteRead != TagConstants.TAG_EBD_END_OF_STREAM
-          && currentByteRead != TagConstants.TAG_EBD_CARRIAGE_RETURN
-          && currentByteRead != TagConstants.TAG_EBD_NEW_LINE) {
-        receivedBytes[receivedBytesInsertIndex] = (byte) currentByteRead;
-        receivedBytesInsertIndex++;
+      try {
+        if (currentByteRead != TagConstants.TAG_EBD_END_OF_STREAM
+            && currentByteRead != TagConstants.TAG_EBD_CARRIAGE_RETURN
+            && currentByteRead != TagConstants.TAG_EBD_NEW_LINE) {
+          receivedBytes[receivedBytesInsertIndex] = (byte) currentByteRead;
+          receivedBytesInsertIndex++;
+        }
+      } catch (ArrayIndexOutOfBoundsException e) {
+        Logger.LOG_EXCEPTION(e);
+        Logger.LOG_CRITICAL(
+            "var_lst.txt has line that exceeds static buffer. Must rebuild with larger"
+                + " maxBytesPerLine in TagInfoManager -> refreshTagList");
+        System.exit(1);
```

Comment:
I don't think there is much else that can be done given the current structures of our connectors, but I am a little concerned with the potential side effects that a straight `System.exit(int);` call makes.

Connectors currently handle shutdown by gracefully stopping any threads, MQTT connections, etc. 
- A call to `System.exit(int);` bypasses that graceful shutdown/cleanup process

In the future, I think it might be good to evaluate how to add a shutdown hook or handler so that errors in the library still can trigger a graceful cleanup/shutdown

Reply:
That's a good point. I guess that we can't always assume this will be called during initialization, before connections are made.  System.exit() will exit the process. Other threads belong to the process will be stopped. The Flexy FW handles connections.  Shutdown hook may be the best way to try gracefully.

Reply:
For this specific issue we can solve this in a non fatal way. But for "Fatal" errors in other libraries I would rather see the library return an error code or some other notification to the application using the library that the library operation failed. That way the application can determine to kill the process or it can attempt to recover or continue in a wounded state. This probably depends on the error encountered in the library but its generally best just to notify the application that the library encountered an error and the application can determine how to continue. 

Somewhat related I was thinking about the logging library this week and having a queue of the last messages (printed and non printed). If the application did have a shutdown method, that method could call Logger.printQueue(). and print a configurable amount of messages the preceded the application actually halting. This would probably be useful for a debugging a application exit that may not be easy to replicate or happened while logs were turned down.